src/commando/core.py

```diff
@@ -1,2 +1,79 @@
-def hello_world():
-    return "Hello, world!"
+import subprocess
+import tempfile
+from functools import partial
+from subprocess import PIPE, STDOUT, CalledProcessError
+
+
+def _pprint(repo, message):
+    print(f"{repo}> {message}")
+
+
+def _run(cwd, results, command, **kwargs):
+    result = subprocess.run(  # pylint:disable=subprocess-run-check
+        command, cwd=cwd, stdout=PIPE, stderr=STDOUT, text=True, **kwargs
+    )
+    results.append(result)
+    result.check_returncode()
+    return result
+
+
+def _process_repo(
+    repo, command, commit_message, branch, pr_title, pr_body
+):  # pylint:disable=too-many-arguments
+
+    # Collect the results of all commands run for this repo.
+    results = []
+
+    pprint = partial(_pprint, repo)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        run = partial(_run, tmpdirname, results)
+
+        try:
+            pprint("cloning")
+            run(["gh", "repo", "clone", repo, tmpdirname, "--", "--quiet"])
+
+            pprint(command)
+            run(command, shell=True)
+
+            if run(["git", "status", "--porcelain"]).stdout:
+                pprint("committing changes")
+                run(["git", "switch", "--force-create", branch])
+                run(["git", "add", "--all"])
+                run(["git", "commit", "--message", commit_message])
+
+                pprint("sending PR")
+
+                try:
+                    pprint(
+                        run(
+                            ["gh-pr-upsert", "--title", pr_title, "--body", pr_body]
+                        ).stdout.rstrip()
+                    )
+                except CalledProcessError as err:  # pragma: no cover
+                    if err.returncode == 1:
+                        raise
+            else:  # pragma: no cover
+                pprint("no changes to commit")
+        except CalledProcessError:  # pragma: no cover
+            # Print the outputs of all the failed repo's commands.
+            for result in results:
+                print(f"$ {result.args} (returncode was: {result.returncode})")
+                print(result.stdout)
+
+            raise
+
+
+def commando(
+    repos, command, commit_message, branch, pr_title, pr_body
+):  # pylint:disable=too-many-arguments
+    failed = False
+
+    for repo in repos:
+        try:
+            _process_repo(repo, command, commit_message, branch, pr_title, pr_body)
+        except CalledProcessError:  # pragma: no cover
+            failed = True
```

Comment:
If processing one repo fails we don't let that crash us and we continue with processing the rest of the repos. This is what we want: we don't want one broken repo to prevent a scheduled GitHub Actions workflow from sending PRs to all the following repos. But we record the fact that at least one repo has failed and exit with status 1 at the end, so for example a GHA workflow run will be marked as failed if any of the repos fail. Again this is what we want: we want to be notified if a repo is failing.